smac/intensification/successive_halving.py

```diff
@@ -19,6 +19,10 @@
 
 
 class SuccessiveHalving(AbstractRacer):
+    curr_inst_idx = 0
+    running_challenger = None
+    curr_challengers = set()  # type: typing.Set[Configuration]
```

Comment:
Why define only 3 of the stage variables outside init?

Reply:
`mypy` is able to infer the types of the other variables. For a `set()` it's unclear what will be in there so I had to define it.

Reply:
Yes, but I was wondering why the three variables here (`curr_inst_idx`, `running_challenger`, `curr_challengers`) are initialized outside of `__init__` as global class variables.

Reply:
It seems that this is not necessary and the variables can be defined in the init, too. Will change.